Gem/Code/Source/Components/Multiplayer/PlayerArmorComponent.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <UiPlayerArmorBus.h>
+#include <Components/NetworkHealthComponent.h>
+#include <Source/Components/Multiplayer/PlayerArmorComponent.h>
+
+namespace MultiplayerSample
+{
+    PlayerArmorComponentController::PlayerArmorComponentController(PlayerArmorComponent& parent)
+        : PlayerArmorComponentControllerBase(parent)
+    {
+    }
+
+    void PlayerArmorComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        if (IsAutonomous())
+        {
+            GetNetworkHealthComponentController()->GetParent().HealthAddEvent(m_changedHandler);
+        }
+
+        if (IsAuthority())
+        {
+            GetNetworkHealthComponentController()->SetHealth(GetStartingArmor());
+        }
+    }
+
+    void PlayerArmorComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        m_changedHandler.Disconnect();
```

Comment:
is there any penalty for calling this if `IsAuthority()` and we haven't registered this event?

Reply:
There is a check for this either way, so I think it is optional as there is always going to be a basic check somewhere. This one might be even faster.
```
    void EventHandler<Params...>::Disconnect()
    {
        if (m_event)
        {
            m_event->Disconnect(*this);
        }
    }
```
Compared to:
```
    bool MultiplayerController::IsAutonomous() const
    {
        return GetNetBindComponent() ? GetNetBindComponent()->IsNetEntityRoleAutonomous() : false;
    }
```